src/cairoUtilFuncGen/event.ts

```diff
@@ -113,12 +111,10 @@ export class EventFunction extends StringIndexedFuncGen {
 
     const cairoParams = params.map((p) => `${p.name} : ${p.type}`).join(', ');
 
-    const topic: BigInt =
-      BigInt(
-        `0x${keccak('keccak256')
-          .update(node.canonicalSignature(ABIEncoderVersion.V2))
-          .digest('hex')}`,
-      ) & BigInt(MASK_250);
+    const topic: string = warpEventCanonicalSignaturehash(
+      node.name,
+      node.vParameters.vParameters.map((param) => param.typeString),
```

Comment:
It may contain `uint` , should be `uint256` instead

Reply:
The typestrings are canonicalised by solc

Reply:
It doesn't work for structs though